app/src/main/java/com/metro/gwuexplorer/AlertManager.kt

```diff
@@ -0,0 +1,82 @@
+package com.metro.gwuexplorer
+
+import android.widget.Toast
+import okhttp3.*
+import okhttp3.logging.HttpLoggingInterceptor
+import org.json.JSONObject
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+class AlertManager {
+    // OkHttp is a library used to make network calls
+    private val okHttpClient: OkHttpClient
+
+    init {
+        val builder = OkHttpClient.Builder()
+
+        // This sets network timeouts (in case the phone can't connect
+        // to the server or the server is down)
+        builder.connectTimeout(20, TimeUnit.SECONDS)
+        builder.readTimeout(20, TimeUnit.SECONDS)
+        builder.writeTimeout(20, TimeUnit.SECONDS)
+
+        // This causes all network traffic to be logged to the console
+        val logging = HttpLoggingInterceptor()
+        logging.level = HttpLoggingInterceptor.Level.BODY
+        builder.addInterceptor(logging)
+
+        okHttpClient = builder.build()
+    }
+
+
+
+    fun retrieveAlert(
+        successCallback: (List<Alert>) -> Unit,
+        errorCallback: (Exception) -> Unit
+    ) {
+        // Data setup
+        val primaryKey = "da0eb4c222b5442ea5a2a59b3677b73e"
```

Comment:
Uh oh -- make sure you hide this...

This is also a public repo, so I'd also suggest regenerating your key, just in case...

Reply:
At the end of Lecture 7, I have some slides with ideas of how you can hide your metro key